### ISSUE TITLE
HTTPS Support

### ISSUE THREAD

**Issue by spencerbeggs** [getumbrel/umbrel#546]
Communicating with the node over HTTP on your local network seems dicey. Would you be interested in a pull request that would generate SSL certs locally and then allow you to download the root cert from the settings menu? I might be able to hack that together.

**Comment by lukechilds** [comment]
Thanks for the suggestion!

During beta, Umbrel makes the assumption that the local network is secure. This is pretty much the industry standard and how every consumer router or smart device that exposes a web interface work.

We agree this isn't really good enough as an industry standard. We think we have some clever ideas on how we can do secure communication out of the box for a stable release. However I think it's out of scope for the beta.

Also, I don't think encouraging users to install root certs is a good idea. That should only be done by very technical users who know what they're doing and understand the implications. As an aside, if you're concerned your local network is not secure then anything malicious could inject a fake root cert when you download it over HTTP, and then compromise your entire browser/OS when you install the bad cert.

For now, if you're worried about plain text local network communication, I'd recommend accessing your Umbrel via the Tor hidden service which will ensure all data is encrypted in transit.

**Comment by tim-tx** [comment]
I would vote to re-open this, not because of unencrypted data in the local network but because of unencrypted data in Tor. If I understand Tor correctly, then without HTTPS, traffic in Tor is only encrypted up to the exit node. No extra encryption exists between the exit node and the destination. There was a prominent attack on Tor traffic exploiting this in 2007. Without HTTPS, the exit node or some eavesdropper between the exit node and the destination could very easily read my Umbrel password. Likewise, if I use the Electrum server without HTTPS over Tor, my wallet information would be totally exposed between the exit node and the destination. Why not use a self-signed certificate, since I am both the user and the certificate creator?

**Comment by tim-tx** [comment]
Ah, I found this is discussed in #190 and irrelevant since you don't ever exit to clearnet.

**Comment by djkazic** [comment]
Hello, just wanted to chime in with my two cents. I've got good results in modifying my nginx.conf and using certbot + dns01 challenges to get SSL certs for use locally. My dns records just point at the private IP so nothing touches clearnet while getting that nice green checkmark in the browser

**Comment by BenGWeeks** [comment]
Adding my vote for this one. This is required to run BTCPayServer I believe.

**Comment by abradshaw** [comment]
Agreed, this really needs to be addressed

**Comment by BenGWeeks** [comment]
For following instructions could be a useful resource (specific to BTCPayServer but presumably could be generalised):

[Installing the NGINX reverse proxy with an SSL certificate for Umbrel / BTCPay Server](https://jorijn.com/installing-nginx-reverse-proxy-with-ssl-certificate-umbrel-btcpayserver/)

**Comment by BenGWeeks** [comment]
Why was this closed?

**Comment by KayBeSee** [comment]
Just noting that enabling access via HTTPS would allow browsers to access and use camera functionality. 

See https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia

**Comment by marcelrv** [comment]
FYI issue you will have related to this topic: https://community.getumbrel.com/t/apps-remain-in-starting-but-actually-work-fine/6338/7

**Comment by jbrill** [comment]
All it takes is one Umbrel vuln to bring down half of the lightning network?

HTTPS integration needs to be the #1 priority for the Umbrel team at this point. As the network continues to scale, it is unacceptable to assume that everyone's local network should be secure, especially when that service communicates with the outside world.

**Comment by relativisticelectron** [comment]
Yes, why closed?  An umbrel https connection would also be a benefit for [specter](https://github.com/cryptoadvance/specter-desktop/issues/778), because it allows [Notifications_API](https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API/Using_the_Notifications_API).

**Comment by i5hi** [comment]
This makes running a home node via Tor insecure. Self signed cert is fine for private use. Seems like a basic requirement.

**Comment by simplismio** [comment]
Besides that it insecure for Tor, Apple will soon block API calls shortly (currently depreciated), making impossible for apps like Zeus and others to connect to Umbrel over Tor. Currently, NSAllowsArbitraryLoads -> true is required to make the connection possible.

**Comment by yavko** [comment]
HTTPS is required for some browser api's breaking some things in nextcloud, could this please be re-opened?

**Comment by artizzle** [comment]
> HTTPS is required for some browser api's breaking some things in nextcloud, could this please be re-opened?

I want https because of nextcloud as well. With just the tor I can't use the nextcloud app.

**Comment by lukechilds** [comment]
Happy to re-open this issue to keep track of it.

Since there's been a lot of discussion here recently I'll link to the places this has been brought up before. If you're wondering why Umbrel doesn't yet support SSL, it's because it's not trivial to support SSL over the local network in a way that doesn't fail to actually prevent MITM attacks or introduce dangerous security footguns, you can read more on the reasons why in these previous discussions:

- https://github.com/getumbrel/umbrel-os/pull/53#issue-661821443
- https://github.com/getumbrel/umbrel/issues/190#issuecomment-683626245
- https://github.com/getumbrel/umbrel/issues/985#issuecomment-921041368

We definitely want to support this at some point, and we have some ideas to experiment with, but it’s not a simple fix and we want to devote some more time in the future to make sure whatever solution we implement overcomes the existing shortcomings of using SSL on the local network.

**Comment by prologic** [comment]
I just installed a test version of Umbrel and to my surprise I also discovered that the default configuration is **insecure**:

```#!sh
root@umbrel:~# ss -tapn | grep LISTEN
LISTEN   0         4096                0.0.0.0:80               0.0.0.0:*        users:(("docker-proxy",pid=49619,fd=4))
```

This effectively means ingress traffic from "anywhere" to Umbrel can be sniffed.

Assuming the "local" network is "secure" is foolish and just plain wrong.

At a minimum adding Let's Encrypt support would be the easiest way to fix this.

**Comment by banneord-puzzle** [comment]
Umbrel without SSL is a toy. I just started to test this software stack and already want to move to other solutions because it's incomprehensible to me, how can Umbrel be a serious proposition for a Bitcoin/LN node without proper encryption.

The assumption about safe network is just plain wrong. The tale about unsolvable problem of MITM on local networks is preposterous. If Umbrel image has to be written to SD card then it's a no brainer to generate/add certs at this point to the card as well.

As mentioned by @jbrill - all it takes to bring down Umbrel is one exploit that can be automated and as Cave Johnson said it - "We're done here". Lack of SSL and no seed-based-non-default-passwords is IMHO simply reckless.

**Comment by connected201** [comment]
Add SSL, without this, nobody use umbrel for production, and TOR is not convient for speed and specific browser requirement.
Take care about security is most important that you need to do!

**Comment by connected201** [comment]
i able to fix problem using https://github.com/suyashkumar/ssl-proxy

**Comment by marcelrv** [comment]
I indeed use something similar but ,based on nginx.
What would be a super add-on is a way to create a nginx config file with all the proper ports and forwards defined (e.g based on the .env file)

**Comment by yavko** [comment]
There should be a gui for this kinda like the nextcloud cli tool for adding ssl/tls

**Comment by tlindi** [comment]
There's a nice proposal for fixing this:

https://makers.bolt.fun/story/easy-switch-tor-clearnet-for-bundle-nodes--155

**Comment by justo4** [comment]
> i able to fix problem using https://github.com/suyashkumar/ssl-proxy

Did you build it on your Umbrel node (putting golang on there etc) or were you able to get "docker-compose -f ..." to work?

**Comment by connected201** [comment]
no, i run ssl proxy and connect to web interface over ssl proxy, rad documentation, just download ssl proxy and run like this `ssl-proxy -from 0.0.0.0:4430 -to 127.0.0.1:8000` where `127.0.0.1:8000` is dockerw web interface for casaos or other container

**Comment by ryenski** [comment]
This should be possible using Tailscale: https://tailscale.com/kb/1153/enabling-https/
However, Umbrel does not expose port 443 by default.

**Comment by habibitcoin** [comment]
Adding on, definitely believe this is crucial

**Comment by tlindi** [comment]
> This should be possible using Tailscale: https://tailscale.com/kb/1153/enabling-https/ However, Umbrel does not expose port 443 by default.

Please, no additional components and service just to expose one application port.

**Comment by ryenski** [comment]
> Please, no additional components and service just to expose one application port.

Tailscale is already included.

**Comment by tlindi** [comment]
And it does not increase number of software nor service providers and costs to run node? And it is improving decentralization on node runners. Please?

**Comment by BenGWeeks** [comment]
Tailscale is a workaround, but bad UX when users have to remember to run it to use their wallet connected to their node.

**Comment by Syturio** [comment]
Can some1 explain detailed how can we get HTTPS via Tailscale

**Comment by coolaj86** [comment]
Re: https://github.com/getumbrel/umbrel/issues/1576#issuecomment-1380798757

@lukechilds Caddy does the client-provisioning as well. It's literally a pop-up to add a cert to your keychain.

That aside, none of those arguments hold water in the least.

That's like saying "We're not going to give you a knife because someone else might show up with a gun, or even a bigger knife. Best to be unarmed, and naked."

These are not challenging problems in today's age. No one is using SSL anymore. Dynamic DNS is free from multiple reputable providers from Name.com and DNSimple to GoDaddy and DuckDNS. Saying that customers can't trust a DNS you provide is a non-argument - they're running _your_ software. They trust you.

This is akin to the 90s mentality of "don't curl | bash" - which I'm sure you are well aware of - all of the technical arguments against doing The Right Thing™ are based in myths that have a basis in (old) reality, but not the reality that we live in today (or for the last 5 years or so).

**Comment by warioishere** [comment]
i would like to open this again. Ist Umbrel anyway still alive or a deadwreck?

**Comment by PilotoCero** [comment]
It's no possible for example to use LNbits (Boltcard extension) because the NFC doesn't work over a non secure network. So we really need https://

**Comment by ladstaetter** [comment]
well his years huge umbrel os update implements only a freaking new skin for the appstore, thats all. i don't even understand why there is a appstore when half of the apps not even working because they need https.

**Comment by aantonop** [comment]
Lack of HTTPS also disables Webauthn security key two-factor authentication in several Umbrel apps, such as BTCPay and Nextcloud

**Comment by rajarshimaitra** [comment]
This has been a long lasted ask and a necessity. We hear this from our local community of user space all the time. Umbrel basically kills the possibility of doing self-hosted secured btcpay server for merchants via their node. If that's not the THE MOST important use case for a node, I don't know what is. 

Hope the Umbrel team considers this.

**Comment by aviv57** [comment]
I'm willing to donate a bounty of 0.01BTC for umbrel to support HTTPS, this is a must have for umbrel to run a trust worthy bitcoin lightning and onchain wallets

**Comment by ztnewman** [comment]
Why not just include a self signed cert by default?

**Comment by Cazza9** [comment]
for self hosting you need https

**Comment by lorenzyannick** [comment]
Maybe umbrel can implement traefik ( https://doc.traefik.io/traefik/ )
With auto-discover. So you can add labels variables in docker to enable proxy with HTTPS over each apps 

a good start can be adding labels in docker files like :


`
labels:
      traefik.enable: true
traefik.http.routers.<service_name>.rule: Host(`${IP_OR_DOMAIN_NAME}`) # Host requested should be IP_OR_DOMAIN_NAME
      traefik.http.routers.<service_name>.entrypoints: http # maybe use https here ?
      traefik.http.routers.<service_name>.service: ${CONTAINER_NAME}
      traefik.http.services.<service_name>.loadBalancer.server.port: 80 # Request redirect to that port of host : maybe use 443 here...
`

**Comment by RuneStone0** [comment]
+1 please find a solution for this!  Looks like there are several good suggestions how to resolve this already ([example](https://github.com/getumbrel/umbrel/issues/546#issuecomment-1304562072))

**Comment by emileond** [comment]
+1 we need HTTPS

**Comment by TWP80** [comment]
**i need https connection for btcpay server connect to woocommers any idea how can i do this.**

**Comment by guttermonk** [comment]
> **i need https connection for btcpay server connect to woocommers any idea how can i do this.**

You can either wait for this issue to be closed (and it's been open since 2021), or you could get an Embassy from Start9.

**Comment by tlindi** [comment]
Also MyNodeBTC Premium (some ~$99) works too,

**Comment by rhubbard-nwf** [comment]
It's concerning that the devs believe the "local net" should be trusted, and so SSL is not a SUPER HIGH PRIORITY... I can't run LND in any form of production without proper security, local net included/especially. The security rule is simple... NOTHING IS TRUSTED. Trustless networking IS the industry standard now, and has been for 2+ years...

I really hope this issue gets resolved. This is reckless.

**Comment by highghlow** [comment]
+1

**Comment by ljacho** [comment]
+1

**Comment by JPLACLAU** [comment]
+1 just implement https the whole world is asking for it and needing it.

**Comment by t0m7** [comment]
+1

**Comment by CelestialV31L** [comment]
+1 please

**Comment by dinkelk** [comment]
+1

**Comment by SqueakyBed** [comment]
+1³x24ⁿ

**Comment by DavidGarciaCat** [comment]
I just wanted to add two small tips for this thread:

- one option to "expedite" the HTTPS support is to install the CloudFlare Tunnel app from the Umbrel App Store, and then create a tunnel for that
  - as part of the local LAN, you will keep an insecure connection if you target the device IP address or the device name
  - but you can "by-pass" that insecure connection by using the CloudFlare Tunnel DNS you have created

- another point is that HTTPS seems a must-have if we want to enable TOR access to the Umbrel device
  - I have just tested it, and the ONION domain is accessible over HTTP instead of HTTPS
  - in a case like this, the CloudFlare Tunnel won't solve the problem

**Comment by sahilph** [comment]
I have created Nginx Proxy Manager App for Umbrel, available on [my community App Store](https://github.com/sahilph/sahil-umbrel-apps). I have also created  a pull request [#1296](https://github.com/getumbrel/umbrel-apps/pull/1296) to add it in Umrel App Store.

While this may not solve all the security issues discussed, it can help to access certain apps  that require HTTPS

**Comment by yevmoroz** [comment]
I needed HTTPS too. Some apps would make it critical requirement.

I tried creating apps based on nginx and caddy for Umbrel appstore, but that's too heavy for a small pi4.

I end up doing something far more interesting and available nearly out of the box.
Because I'm using Tailscale already to extend my network.
Here is a solution (you only have to do this once):

- `sudo docker exec -it tailscale_web_1 sh` to get that container with
  - I have name be default `tailscale_web_1`
  -  if you have different, run `sudo docker ps | grep tailscale/tailscale` to find NAME
  - once inside, you are exposed to run many commands with `tailscale` executable
- `tailscale cert` to generate certificates 
  - Tailscale needs it's own certificates to serve HTTPS content (duh)
  - make sure HTTPS support enabled in Tailscale "admin console" as well and you have "tailnet"
  -  you can read about it [here](https://tailscale.com/kb/1153/enabling-https)
- `tailscale serve --bg --https=000 8123` for homeassisstant
  - `8123`  is default port used by homeassisstant
  - `000` choose number on your own, remember that SSL is by default 443
  - again, [doc](https://tailscale.com/kb/1242/tailscale-serve) on this command for more info
 -  `tailscale serve --bg --https=000 2283` for immich
 - and so on for more services
 
This will make it so that you can access these services from within the Tailnet only but via URL with HTTPS support. For example `https://tail000000.ts.net:000` obviously zeroes replaced with valid numbers that are specific to you and your Tailscale setup.

Alternative if you want to expose it to the world (everyone outside your Tailnet, NOT RECOMMENDED) you could look into `tailscale funnel`, it has some similar things. [doc](https://tailscale.com/kb/1311/tailscale-funnel)

**Comment by jjmmbb** [comment]
3 years and nothing to support HTTPS...

**Comment by wh0am1-dev** [comment]
@sahilph, I've been unable to properly expose apps through Nginx Proxy Manager. Not even using HTTP, so configuring HTTPS is completely out of reach for me.

<details><summary>Steps I followed:</summary>

Let's say my domain is `example.com`.

- Pointed my domain root (`@` A record) to my public IP
- Pointed all subdomains (`*` A record) to my public IP
- Forwarded external port `80` (HTTP) to Umbrel's port `40080` (NPM HTTP port exposed through `app_proxy`)
- Forwarded external port `443` (HTTPS) to Umbrel's port `40443` (NPM HTTPS port exposed through `app_proxy`)
- Created a Proxy Host in NPM to forward subdomain `up.example.com` to `http://umbrel.local:8385` (Uptime Kuma), SSL deactivated
- When trying to access `up.example.com` I get redirected to Umbrel's auth interceptor `up.example.com:2000`
- As this wasn't even loading, I tried forwarding external port `2000` to Umbrel's port `2000`
- Then, Umbrel's auth interceptor loaded, but when typing the password and hitting enter, that page doesn't do anything (checked out the browser's network tab in devtools and there were a bunch of errored requests...)

I also read through NPM app submission PR comments and saw Umbrel devs and @sahilph talking about using different hosts for Proxy Hosts inside NPM instead of `umbrel.local`: app docker container name, `uptime-kuma_server_1`, or internal Umbrel's gateway IP, `10.21.0.1`... but none of those work either.

Also tried routing to apps using paths instead of subdomains, eg. `example.com/up`, because I thought that maybe the auth interceptor stores whatever token linked only to the specific domain you do login, and not its subdomains... I tried everything I could come up with, honestly. Today I'll give it another round of tries to see if I missed anything, but I've already spent hours trying to set this up properly to no luck.

</details>

I'm really stuck with this and Umbrel supporting HTTPS by default would be more than ideal, I'd say even necessary! @mayankchhabra @lukechilds @nmfretz

**Comment by sahilph** [comment]
@wh0am1-dev Nginx Proxy manager is ideally meant to be used for apps that have their own authentication and are not protected by Umbrel's auth server (Like Jellyfin, Plex, etc ).

It is not currently possible to use NPM with  Umbrel's auth server. So if you want to use NPM for apps which are protected by auth server, you will have to bypass the auth server and by pointing NPM to the app's _internal_ hostname/ip and _internal_ port.

⚠️ Bypassing the auth server will make the app accessible to anyone. Hence, it recommended to use the NPM "_Access Lists_" and setup Username/Password for that app. ⚠️

>I also read through NPM app submission PR comments and saw Umbrel devs and @sahilph talking about using different hosts for Proxy Hosts inside NPM instead of umbrel.local: app docker container name, uptime-kuma_server_1, or internal Umbrel's gateway IP, 10.21.0.1... but none of those work either.

You almost got that right, but instead of gateway IP, you will have the app's _internal_ hostname/ip and _internal_ port.

In your case, for uptime kuma:
 _internal_ hostname: uptime-kuma_server_1
_internal_ port: 3001

>Created a Proxy Host in NPM to forward subdomain up.example.com to http://umbrel.local:8385 (Uptime Kuma), SSL deactivated

So basically you will have to change it to http://uptime-kuma_server_1:3001

**Comment by henriquesebastiao** [comment]
+1

**Comment by NuiceT** [comment]
+1

**Comment by justvanbloom** [comment]
+1

**Comment by sahilph** [comment]
>Thank you @sahilph — are you aware if this is also true for users running Umbrel behind pfSense or HAProxy (or some other firewall + reverse proxy)?

For apps having their own authentication, yes it will work similarly.

However, for apps that use Umbrel's auth server,  _external_ reverse-proxy wont work, as it will not be able to resolve _internal_ hostnames. You will still have to use NPM in that case.

**Comment by jebalert** [comment]
Running this without https is a huge risk.

**Comment by DavidGarciaCat** [comment]
Hey @a-nevermind 

Please don't get me wrong, as I also requested HTTP support to move forward 🙂

All what I say is that there are alternatives we can consider in the short term while waiting for proper HTTPS support

Cheers!

**Comment by justresisting** [comment]
I have spent quite a few hours investigating the pros and cons of running various Homelab OSes. TrueNAS and UNraid were too complicated, HexOS is overcharged, Proxmox isn't intuitive for me, Cloudron costs money, and other options like Sandstorm.io & YunoHost just didn't feel right. That left two choices.. CasaOS/ZimaOS and UmbrelOS. With ZimaOS being based in Shanghai outside of US legal control UmbrelOS was the clear choice for me because (I believed) it does everything that I wanted it too. 

I've been tinkering around with Umbrel OS for a few weeks now. I have been amazed with how easy it has been to install apps; 'one click install' really does mean 'one click'. Up until yesterday I have absolutely been in love with this software. 

It's beautiful, fast, intuitive, & really should be the OS/device of choice for homelab builds.

Except for one thing.. lack of HTTPS. 

It is absolutely inexcusable that HTTPS is not included by default either as an opt-in or opt-out option. Connecting through Tor is not an acceptable alternative. I have read posts on the Umbrel Community pages, Github Issues, watched multiple youtube videos, and there is not a single compelling argument that has been made for the exclusion of HTTPS as a feature.

I was willing to figure out how to run NGINX proxy manager to secure my Umbrel.. until I put an external drive into one of my raspberry pi 5 USB ports which is the device that I installed Umbrel OS on.. 

and.. it wasn't there. 

I could have dealt with finding workarounds for basic security but seeing that [Umbrel has such a disdain for people running their operating system on hardware that isn't there's](https://community.umbrel.com/t/why-did-umbrel-disabled-external-devices-storage-for-raspberry-devices/21613/4) is what motivated me to write this post and has influences my tone. 

Umbrel, for the most part, is an absolutely stellar project with an immense potential to benefit the lives of its userbase by offering an outstanding product AND benefit their company through sales of that product. However under no circumstances would I invest a single dollar into buying that product when the Umbrel userbase has literally been begging for the most basic levels of security for FOUR YEARS. 

There is no compelling technical argument to be made for not including HTTPS out of the box. 

None. 

The refusal to include HTTPS features for four years hasn't just left the project less secure overall but without advanced technical workarounds half of the apps don't even function properly. 

Unacceptable. 

I would have paid 25 bucks a year for a license that would have given me advanced security features like HTTPS! 

Unfortunately without a roadmap that places securing it's devices as it's number 1 goal then this project lacks the necessary credibility for me to invest money into it. 
 
I am not abandoning this project. I want Umbrel to succeed.

The foundation of the project is very strong. Individuals like @sahilph and denny (who oversees community app store) are doing incredible things. However Umbrel's development team needs to earn the trust of it's users by implementing the most basic level of security, HTTPS, into the OS.

I sincerely hope this project considers doing that. Thank you.

**Comment by millettjon** [comment]
I just purchased an Umbrel Home.
Unfortunately, I will never be able to use it given the security posture. Straight to garbage.

**Comment by DavidGarciaCat** [comment]
> I just purchased an Umbrel Home. Unfortunately, I will never be able to use it given the security posture. Straight to garbage.

Until HTTPS support is ready, you can add a reverse proxy with a generic SSL certificate and target the non-secure protocol

However, this will only work for those cases where UmbrelOS Auth Proxy doesn't mess up with the services you want to run

Alternatively, if you are a tech-y person, you can try to get rid of UmbrelOS and install your own choice of OS, and then run all the services you want/need (no matter if they are available in the UmbrelOS App Store or not)

Spoiler alert: I hacked my UmbrelHome because I didn't want to wait any longer for specific updates, and now I have an UmbrelHome but it's not running UmbrelOS anymore (as it should always be, given I paid for the hardware, and no one should be tied to stick with the pre-installed OS - especially being Linux and running free-of-charge Docker Apps)

**Comment by jjmmbb** [comment]
It’s been years since this issue was first raised, and the lack of HTTPS support on Umbrel, even on a local network, has gone from a minor inconvenience to a completely unacceptable security gap.

At this point, the arguments for postponing HTTPS simply don’t hold up. The ecosystem has evolved, the threat landscape has evolved, and Umbrel itself has evolved. What hasn’t evolved is the project’s stance on the most basic security requirement of any modern web service: encrypted transport.

Relying on the assumption that a local network is “secure” is not just outdated, it’s reckless. Local networks get compromised. Malware spreads laterally. Users connect IoT devices with terrible security. Attackers exploit ARP spoofing, DNS poisoning, and packet sniffing with trivial tools. Pretending that HTTP is “good enough” because it’s behind a router is a security posture from another decade.

And beyond the security implications, the lack of HTTPS breaks functionality. Many Umbrel apps require HTTPS to operate correctly. Modern browsers block features like camera access, service workers, notifications, and other APIs unless the page is served over HTTPS. Users are forced into hacky workarounds, reverse proxies, or running duplicate stacks outside Umbrel, all because the platform refuses to implement something that should be standard.

The community has repeatedly expressed the need for this. Multiple users have provided working examples, proposed solutions, and even demonstrated that self‑signed certificates or local CA generation can be implemented safely. Other homelab platforms solved this years ago. Umbrel is now the outlier, and not in a good way.

This issue has been closed without resolution more than once, despite ongoing user demand and clear evidence that the current approach is insufficient. It’s time to acknowledge that this is not a “nice to have” feature. It’s a fundamental requirement for a platform that handles sensitive data, financial information, authentication tokens, and network‑exposed services.

I am suggesting that this issue be reopened and that HTTPS support, even if optional, be prioritized. Four years of delay is long enough. The community deserves a clear, actionable roadmap and a commitment to delivering secure-by-default communication.

Umbrel cannot continue to ignore this. HTTPS is not optional. It is not “advanced.” It is not something users should have to duct‑tape themselves. It is the baseline for any system that expects to be taken seriously.

**Comment by millettjon** [comment]
Unfortunately, due to the security posture, the entire project is suspect.

**Comment by jjmmbb** [comment]
> Unfortunately, due to the security posture, the entire project is suspect.

How can passwords used locally only can be listed as leaked passwords? I am forced to agree with you.

**Comment by jjmmbb** [comment]
> Unfortunately, due to the security posture, the entire project is suspect.

It looks like the hardest part of this work is already done. Someone in the community has rewritten the app‑proxy code to support HTTPS, which makes things much easier to integrate officially into the project.

Since there’s now a working implementation that proves the concept, the next steps should mostly involve reviewing the changes, aligning them with the project’s standards, and deciding how to merge them cleanly. This puts us in a much better position to move HTTPS support forward without having to start from scratch.

If there’s anything specific that still needs to be evaluated or tested, feel free to point it out, happy to help move this along.

**Comment by sahilph** [comment]
> It looks like the hardest part of this work is already done. Someone in the community has rewritten the app‑proxy code to support HTTPS, which makes things much easier to integrate officially into the project.


Yes, I had written[ this app](https://github.com/sahilph/umbrel-HTTPSizer) a while back, it would allow all the apps that use `umbrel-app-proxy`, including the main umbrel dashboard, to be accessible via HTTPS using self signed certificate.
